src/service/extension-location.js

```diff
@@ -59,7 +60,14 @@ export function calculateExtensionScriptUrl(
   opt_extensionVersion,
   opt_isLocalDev
 ) {
-  const fileExtension = getMode().esm ? '.mjs' : '.js';
+  let fileExtension;
```

Comment:
Consider making this blurb a helper `getFileExtension` given that it's doing more work than `finalizeScriptUrl`

Reply:
+1, there's already repetition in this function and the one below